app/models/service_terraform_template.rb

```diff
@@ -75,8 +76,40 @@ def check_refreshed(_action)
 
   private
 
+  def job(action)
+    stack(action).miq_task.job
+  rescue err
+    _log.error("Could not fetch #{action} Job", err)
+    nil
+  end
+
   def get_job_options(action)
-    options[job_option_key(action)].deep_dup
+    job_options = options[job_option_key(action)].deep_dup
+
+    # Add additional vars,
+    # :miq_action                    -> required for Retirement action
+    # :miq_terraform_runner_stack_id -> Terraform-Runner stack_id required by Retirement action.
+
+    case action
+    when ResourceAction::RETIREMENT
```

Comment:
The `case` with only one state is odd, could you just do `if action == ResourceAction::RETIREMENT`